src/nl/hannahsten/texifyidea/inspections/latex/probablebugs/LatexSuspiciousSectionFormattingInspection.kt

```diff
@@ -0,0 +1,80 @@
+import nl.hannahsten.texifyidea.util.containsAny
+import nl.hannahsten.texifyidea.util.files.commandsInFile
+import nl.hannahsten.texifyidea.util.firstChildOfType
+import nl.hannahsten.texifyidea.util.magic.CommandMagic
+import nl.hannahsten.texifyidea.util.requiredParameter
+
+open class LatexSuspiciousSectionFormattingInspection : TexifyInspectionBase() {
+
+    override val inspectionGroup = InsightGroup.LATEX
+
+    override fun getDisplayName() = "Suspicious formatting in the required argument of a sectioning command"
+
+    override val inspectionId = "SuspiciousSectionFormatting"
+
+    override fun inspectFile(file: PsiFile, manager: InspectionManager, isOntheFly: Boolean): List<ProblemDescriptor> {
+        return file.commandsInFile()
+            .asSequence()
+            .filter { it.name in CommandMagic.sectionMarkers }
+            .filter { it.optionalParameterMap.isEmpty() }
+            .filter { it.requiredParameter(0)?.containsAny(formatting) == true }
+            .map { psiElement ->
+                val requiredParam = psiElement.firstChildOfType(LatexRequiredParam::class)
+                // Plus 1 for the opening brace.
+                val startOffset = requiredParam?.startOffsetIn(psiElement)?.plus(1) ?: 0
+                // Minus 2 for the braces surrounding the parameter.
+                val endOffset = requiredParam?.textLength?.minus(2)?.plus(startOffset) ?: psiElement.textLength
+                manager.createProblemDescriptor(
+                    psiElement,
+                    TextRange(startOffset, endOffset),
+                    "Suspicious formatting in ${psiElement.name}",
+                    ProblemHighlightType.WARNING,
+                    isOntheFly,
+                    AddOptionalArgumentQuickFix()
+                )
+            }
+            .toList()
+    }
+
+    class AddOptionalArgumentQuickFix : LocalQuickFix {
+
+        override fun getFamilyName(): String {
+            return "Add optional argument without formatting"
+        }
+
+        override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+            val command = descriptor.psiElement as LatexCommands
+            val requiredParamText = command.requiredParameter(0)
+            val optionalParamText = requiredParamText?.replace(Regex(formatting.joinToString("", prefix = "[", postfix = "]")), " ")
+                ?: return
+            val optionalArgument = LatexPsiHelper(project).createOptionalParameter(optionalParamText)
+
+            command.addAfter(optionalArgument, command.commandToken)
+            // Create a new command and completely replace the old command so all the psi methods will recompute instead
+            // of using old values from their cache.
+            val newCommand = LatexPsiHelper(project).createFromText(command.text).firstChildOfType(LatexCommands::class)
+                ?: return
+            command.parent.replace(newCommand)
```

Comment:
I think you are replacing the LatexNoMathContent by a LatexCommands here, wouldn't it make more sense to replace the LatexCommands with the LatexCommands? Currently this works because LatexNoMathContent can contain only one command, but I wouldn't rely on it for the future